sheepdog/utils/__init__.py

```diff
@@ -437,86 +544,31 @@ def proxy_request(project_id, uuid, data, args, headers, method, action, dry_run
         return flask.Response(json.dumps({'message': message}), status=200)
 
     if action in ['upload', 'initiate_multipart']:
-        update_state(node, UPLOADING_STATE)
+        indexd_obj = set_indexd_state(node.node_id, s3_url, UPLOADING_STATE, indexd_client)
```

Comment:
Can probably just send the indexd_obj instead of making another indexd.get request.